src/main/java/org/opensearch/forecast/model/ForecastTask.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.forecast.model;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.timeseries.annotation.Generated;
+import org.opensearch.timeseries.model.DateRange;
+import org.opensearch.timeseries.model.Entity;
+import org.opensearch.timeseries.model.TimeSeriesTask;
+import org.opensearch.timeseries.util.ParseUtils;
+
+import com.google.common.base.Objects;
+
+public class ForecastTask extends TimeSeriesTask {
+    public static final String FORECASTER_ID_FIELD = "forecaster_id";
+    public static final String FORECASTER_FIELD = "forecaster";
+    public static final String DATE_RANGE_FIELD = "date_range";
+
+    private Forecaster forecaster = null;
+    private DateRange dateRange = null;
+
+    private ForecastTask() {}
+
+    public ForecastTask(StreamInput input) throws IOException {
+        this.taskId = input.readOptionalString();
+        this.taskType = input.readOptionalString();
+        this.configId = input.readOptionalString();
+        if (input.readBoolean()) {
+            this.forecaster = new Forecaster(input);
+        } else {
+            this.forecaster = null;
+        }
+        this.state = input.readOptionalString();
+        this.taskProgress = input.readOptionalFloat();
+        this.initProgress = input.readOptionalFloat();
+        this.currentPiece = input.readOptionalInstant();
+        this.executionStartTime = input.readOptionalInstant();
+        this.executionEndTime = input.readOptionalInstant();
+        this.isLatest = input.readOptionalBoolean();
+        this.error = input.readOptionalString();
+        this.checkpointId = input.readOptionalString();
+        this.lastUpdateTime = input.readOptionalInstant();
+        this.startedBy = input.readOptionalString();
+        this.stoppedBy = input.readOptionalString();
+        this.coordinatingNode = input.readOptionalString();
+        this.workerNode = input.readOptionalString();
+        if (input.readBoolean()) {
+            this.user = new User(input);
+        } else {
+            user = null;
+        }
+        // Below are new fields added since AD 1.1
+        if (input.available() > 0) {
+            if (input.readBoolean()) {
+                this.dateRange = new DateRange(input);
```

Comment:
For something like `dateRange` are you keeping it here for bwc with the shared config and job indices?